### DISCUSSION TITLE
支持多窗口 Tab 及自定义菜单

### DISCUSSION THREAD

**Discussion by normal-coder** [tw93/Pake#475] [Ideas]
### Search before asking

- [X] 我在 [issues](https://github.com/tw93/Pake/issues?q=) 列表中搜索，没有找到类似的内容。
I searched in the [issues](https://github.com/tw93/Pake/issues?q=) and found nothing similar.

### Motivation

目前打包出来的应用是一个单窗口应用。对功能复杂的网站而言，场景上略微薄弱。希望支持支持多窗口 Tab 及自定义菜单，以便于更加灵活的使用。

### Solution

- 多窗口 Tab 支持：参考石墨一类 Electron 应用模式。
- 自定义菜单：考虑开放一个 `config.json` 用于支持自定义菜单结构。支持如下能力：
    - 自定义菜单结构
    - 支持绑定绑定快捷键
    - 快速打开指定 URL 页面（可选开关：当前窗口 Tab 或新窗口 Tab）
    -  提供可供页面 JS 调用的方法，用于扩展菜单选项

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

**Comment by tk103331** [comment]
自定义菜单,这个特性得要做插件系统了，要做的话得好好设计

**Comment by tw93** [comment]
转移到讨论，这个功能需要想想怎么支持更好
